crates/networking/p2p/types.rs

```
@@ -91,14 +93,114 @@ impl Node {
+                "tcp" => decoded_pairs.tcp_port = u16::decode(&value).ok(),
+                "udp" => decoded_pairs.udp_port = u16::decode(&value).ok(),
+                "secp256k1" => {
+                    let Ok(bytes) = Bytes::decode(&value) else {
+                        continue;
+                    };
+                    if bytes.len() < 33 {
+                        continue;
+                    }
+                    decoded_pairs.secp256k1 = Some(H264::from_slice(&bytes))
+                }
+                _ => {}
+            }
+        }
+
+        decoded_pairs
+    }
+
+    pub fn from_node(node: Node, seq: u64, signer: &SigningKey) -> Result<Self, String> {
+        let mut record = NodeRecord {
+            seq,
+            ..Default::default()
+        };
+        record
+            .pairs
+            .push(("id".into(), "v4".encode_to_vec().into()));
+        record
+            .pairs
+            .push(("ip".into(), node.ip.encode_to_vec().into()));
+        record.pairs.push((
+            "secp256k1".into(),
+            signer
+                .verifying_key()
+                .to_encoded_point(true)
+                .as_bytes()
+                .encode_to_vec()
+                .into(),
+        ));
+        record
+            .pairs
+            .push(("tcp".into(), node.tcp_port.encode_to_vec().into()));
+        record
+            .pairs
+            .push(("udp".into(), node.udp_port.encode_to_vec().into()));
+
+        match record.sign_record(signer) {
+            Ok(sig) => record.signature = sig,
+            Err(e) => return Err(e),
+        };
+
+        Ok(record)
+    }
+
+    fn sign_record(&mut self, signer: &SigningKey) -> Result<H512, String> {
+        // note: v is ignored
+        let digest = &self.get_signature_digest();
+        let Ok((signature, _)) = signer.sign_prehash_recoverable(digest) else {
+            return Err("Could not sign record.".to_string());
+        };
```

Nit. It is always better not to discard the original error (again, mostly).
```suggestion
        let (signature, _recovery_id) = signer
            .sign_prehash_recoverable(digest)
            .map_err(|err| format!("Could not sign record: {err}"))?;
```

---

Solved in [51eb231](https://github.com/lambdaclass/ethrex/pull/1728/commits/51eb2310c7b497f5b3290b37a6bc34356b48c059).